synapse/telepath.py

```diff
@@ -135,9 +135,9 @@ async def getAhaProxy(urlinfo):
             cnfo['client'] = client
 
         try:
-            proxy = await client.proxy(timeout=10)
+            proxy = await client.proxy(timeout=5)
 
-            ahasvc = await proxy.getAhaSvc(host)
```

Comment:
Would it be worth adding a timeout to the `openinfo` call after getting the service info as well?

Reply:
That one should obey the built in telepath connect timeout settings supplied in the metadata ( or defaults ).  I think i'd rather just let that one behave normally rather than applying an arbitrary timeout as an airstrike from outside :)